### PR TITLE
Implement all REST API endpoints (GDEV-1584)

### PR DESCRIPTION
**Pull request by Cito**
This PR implements all specified REST API endpoints.

Now only the capability to check which files belong to a given dataset is missing.

### CODE REVIEW

File: wps/core/repository.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        check_valid: bool = True,
+        token: Optional[str] = None,
+    ) -> Optional[WorkPackage]:
+        """Get a work package with the given ID from the repository.
+
+        In the following cases, the method returns None:
+        - if the work package does not exist
+        - if check_valid is set and the work package has expired
+        - if a token is specified and it does not match
+          with what is stored in the work package
```
[the remark this answers is not in the document]
Comment (KerstenBreuer) [reply wps/core/repository.py:131]: PMO

**Review by KerstenBreuer — CHANGES_REQUESTED** [review]
Just some minor suggestions. Please ignore the PMO flagged comments if you don't feel like addressing them right away.

File: wps/adapters/inbound/fastapi_/routes.py [hunk 2]
```diff
@@ -18,17 +18,156 @@
...
+        raise HTTPException(
+            status_code=403, detail="Not authorized to get the work package."
+        )
+    return WorkPackageDetails(
+        type=package.type,
+        file_ids=package.file_ids,
+        file_extensions=package.file_extensions,
+        created=package.created,
+        expires=package.expires,
+    )
```
[the remark this answers is not in the document]
Comment (Cito) [reply wps/adapters/inbound/fastapi_/routes.py]: Made the names more specific in f63e06f.

File: wps/config.py [hunk 3]
```diff
@@ -16,20 +16,23 @@
...
-
     db_name: str = "work-packages"
 
-    work_packages_collection: str = "workPackages"
-
-
-CONFIG = Config()
+    # the default keys are invalid and only set for creating the example configuration
+    auth_key = "{}"  # should be also SecretStr (in ghga-service-commons)
+    work_package_signing_key = SecretStr("{}")
```
[the remark this answers is not in the document]
Comment (Cito) [reply wps/config.py]: Changed in 7644665.

File: wps/core/repository.py [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+        )
+        work_package = await self._dao.insert(work_package_data)
+        encrypted_token = encrypt(token, user_public_crypt4gh_key)
+        return WorkPackageCreationResponse(id=work_package.id, token=encrypted_token)
+
+    async def get(
+        self,
+        work_package_id: str,
+        check_valid: bool = True,
+        token: Optional[str] = None,
```
[the remark this answers is not in the document]
Comment (Cito) [reply wps/core/repository.py]: Yes, the token is required in the route, i.e. you get a "forbidden" error when not passing a token in the header. The repository method is more common and does not require it, but checks the hash it if it is passed.

File: wps/core/repository.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Cito) [reply wps/core/repository.py]: Changed in 997eb9e.

File: wps/core/repository.py [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+            return None
+        return work_package
+
+    async def work_order_token(
+        self,
+        work_package_id: str,
+        file_id: str,
+        check_valid: bool = True,
+        token: Optional[str] = None,
+    ) -> Optional[str]:
```
[the remark this answers is not in the document]
Comment (Cito) [reply wps/core/repository.py]: Changed in 997eb9e.

**Review by KerstenBreuer — APPROVED** [review]

**Pull request merged**